### PR TITLE
workflows: delete branch if PR is closed

### PR DESCRIPTION
**Pull request by ZhongRuoyu**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by Rylan12** [comment]
Isn't there a setting to do this automatically?

Screenshot 2024-07-14 at 3 29 54 PM

**Comment by p-linnane** [comment]
> Isn't there a setting to do this automatically?

It only works when the branch is merged, not when closed.

**Review by p-linnane — APPROVED** [review]

**Review by MikeMcQuaid — APPROVED** [review]
Great idea!

File: .github/workflows/clean-up-closed-prs.yml [hunk 1]
```diff
@@ -53,3 +53,23 @@ jobs:
         run: |
           echo "::notice ::Cancelling tests on closed PR #$PR"
           gh run cancel "$RUN_ID"
+
+  delete-branch:
+    if: >
+      github.repository_owner == 'Homebrew' &&
+      !github.event.pull_request.merged
+    runs-on: ubuntu-latest
```
Comment (carlocab) [inline .github/workflows/clean-up-closed-prs.yml:61]:
```suggestion
    runs-on: ubuntu-latest
    continue-on-error: true
```
This seems more idiomatic for what you want to do here with `|| true`.

Comment (ZhongRuoyu) [reply .github/workflows/clean-up-closed-prs.yml:61]: Thanks; doing this in #177336

**Pull request merged**